src/Microsoft.TestPlatform.ObjectModel/DataCollector/Events/SessionEvents.cs

```diff
@@ -36,10 +39,86 @@ public SessionStartEventArgs() : this(new DataCollectionContext(new SessionId(Gu
         public SessionStartEventArgs(DataCollectionContext context)
             : base(context)
         {
+            this.Properties = new Dictionary<string, object>();
             Debug.Assert(!context.HasTestCase, "Session event has test a case context");
         }
 
         #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Returns the Session Start Properties currently specified.
+        /// </summary>
+        public IEnumerator<KeyValuePair<string, object>> GetProperties()
+        {
+            return this.Properties.GetEnumerator();
+        }
+
+        /// <summary>
+        /// Returns the property value
+        /// </summary>
+        /// <param name="value">
+        /// Value of the property
+        /// </param>
+        /// <param name="property">
+        /// Name of the property
+        /// </param>
+        public T GetPropertyValue<T>(string property)
+        {
+            ValidateArg.NotNullOrEmpty(property, "property");
+
+            T value;
+            if (this.Properties.TryGetValue(property, out object propertyValue) && propertyValue != null)
```

Comment:
T value = this.Properties.ContainsKey(property) ? (T)this.Properties[property] : default(T);

Reply:
Done